development/architecture/migration-guide/horizontal-migration/migrate-list.md

```diff
@@ -0,0 +1,254 @@
+
+```php
+$helperListConfiguration = $this->get('prestashop.core.bridge.helper_list_configuration_factory')->create(
+    self::TABLE,
+    self::CLASS_NAME,
+    $this->controllerConfiguration
+);
+```
+
+To instantiate this configuration, you need to pass it three arguments:
+
+- The name of the table like in legacy (for our example, it's `gender`)
+- The name of the object model class in legacy (for our example, it's `Gender`)
+- An instance of controller configuration, which is an object
+
+### Define the list of fields
+To add a new field to the list, your controller must use the `addListField` methods from the `AdminControllerTrait`.
+
+This method takes an instance of `FieldInterface` and an instance of `HelperListConfiguration`.And she adds the field to the field list in HelperListConfiguration.
+
+A `FieldInterface` object needs a label and an array of configurations. In this array of configurations, you can define the following options:
+
+- title: which is required and represents the text, will be shown in the list's header
+- align: represents where you want to align the value in the cell
+- class: represents a class which be added to the class html attributes in the cell
+- filter_key: represents the name of the filter key
+- orderby: tell if you want or not to be able to order by this field in your list
+- position: identify a field as a position cell
+- search: enable or disable the search by this field
+- width: define the width for the cell of this field
+
+You can find all the configuration of your field in the legacy controller. You have to copy/paste it at the second `config` attribute of the field class.
+
+> Example for the hypothetic TitleController and for the field `id_gender`:
+>
+
+```php
+$this->addListField(new Field(
+        'id_gender', [
+        'title' => $this->trans('ID', [], 'Admin.Global'),
+        'align' => 'center',
+        'class' => 'fixed-width-xs',
+    ]
+), $helperListConfiguration);
+```
+
+### Define actions to your user
+You have two way to add actions, these two methods are in the `AdminControllerTrait`:
+
+- `addAction` which handle `HeaderToolbarAction`
+- `addActionList` which handle `ListHeaderToolbarAction`, `ListRowAction`, `ListBulkAction`
+
+The `HeaderToolbarAction` represents an action in the header toolbar of the page.
+
+The `ListHeaderToolbarAction` represents an action in the header toolbar of the list.
+
+The `ListRowAction` represents an action in each row of the list.
+
+The `ListBulkAction` represents a bulk action for the list.
+
+To create an user action, you must define it as a label and an array of configurations. In this configuration, you can specify the following options:
+
+- confirm: is the text for the confirmation alert
+- desc: is the text for the title HTML attribute
+- href: the link to the action
+- icon: a name of a PrestaShop icons
+- text: the text in the link for the action
```

Comment:
I would suggest first listing what actions are available (would be nice providing images of where they are). And maybe then mention methods in AdminControllerTrait (or mention related methods next to related action like: 
```The `ListRowAction` (image here) represents an action in each row of the list. You can add it via addActionList method in AdminControllerTrait)```.
Or maybe this even deserves a table defining columns, actions, example images and methods.